src/hash.cpp

```diff
@@ -31,7 +31,7 @@ ustring hash(const size_t size, ustring_view msg, std::optional<ustring_view> ke
     if (result_code != 0)
         throw std::runtime_error{"Hash generation failed"};
 
-    return {result, size};
+    return to_unsigned(static_cast<unsigned char*>(result));
```

Comment:
There's some weirdness in this function (including a [VLA](https://en.wikipedia.org/wiki/Variable-length_array)).  I've pushed some code to clean it up, which changes this around to something nicer that MSVC should be happy with.